File: servidor/servidorTCP/src/main/java/org/dogfeeder/model/Pet.java
package org.dogfeeder.model;

import java.sql.Date;
import java.time.LocalDate;
import java.time.format.DateTimeFormatter;

/**
 * Clase para modelar la mascota asociada a DOG-FEEDER
 */
public class Pet {
    public enum TypeOfSize{small,medium,big}
    private int ID;
    private String name;
    private String breed;
    private String birthday;
    private double weight;
    private TypeOfSize type;
    private boolean vaccineNotify;
    private int vaccineDay;
    private int vaccineMonth;

    /**
     * Constructor
     */
    public Pet(){}

    /**
     * Getter para la propiedad ID
     * @return int Identificador de la mascota
     */
    public int getID() {
        return ID;
    }

    /**
     * Setter para la propiedad ID
     * @param ID Identificador para la mascota
     */
    public void setID(int ID) {
        this.ID = ID;
    }

    /**
     * Getter para la propiedad name
     * @return String Cadena que representa el nombre de la mascota
     */
    public String getName() {
        return name;
    }

    /**
     * Setter para la propiedad name
     * @param name Cadena que representa el nombre de la mascota
     */
    public void setName(String name) {
        this.name = name;
    }

    /**
     * Getter para la propiedad breed
     * @return String Cadena que representa la raza de la mascota
     */
    public String getBreed() {
        return breed;
    }

    /**
     * Setter para la propiedad breed
     * @param breed Cadena que representa la raza de la mascota
     */
    public void setBreed(String breed) {
        this.breed = breed;
    }

    /**
     * Getter para la propiedad birthday
     * @return String cadena con la fecha de nacimiento de la mascota
     */
    public String getBirthday() {
        return birthday;
    }

    /**
     * Getter para la propiedad birthday [Date]
     * @return Date fecha de nacimiento de la mascota
     */
    public Date getBirthdayToDate(){
        DateTimeFormatter fmt = DateTimeFormatter.ofPattern("dd/MM/yyyy");
        return  Date.valueOf(LocalDate.parse(birthday, fmt));
    }

    /**
     * Setter para la propiedad birthday
     * @param birthday Cadena que representa la fecha de nacimiento de la mascota
     */
    public void setBirthday(String birthday) {
        this.birthday = birthday;
    }

    /**
     * Getter para la propiedad weight
     * @return double Valor que determina el peso de la mascota
     */
    public double getWeight() {
        return weight;
    }

    /**
     * Setter para la propiedad weight
     * @param weight Valor que determina el peso de la mascota
     */
    public void setWeight(double weight) {
        this.weight = weight;
    }

    /**
     * Getter para la propiedad type
     * @return type Tipo de tamaño segun la raza
     */
    public TypeOfSize getType() {
        return type;
    }

    /**
     * Setter para la propiedad type
     * @param type Tipo de tamaño segun la raza
     */
    public void setType(TypeOfSize type) {
        this.type = type;
    }

    /**
     * Setter para la propiedad type
     * @param type Tipo de tamaño segun la raza
     */
    public void setType(String type){
        this.type = convertType(type);
    }

    /**
     * Método convertType
     * @param type Tipo en cadena de caracteres
     * @return TypeOfSize Valor equivalente convertido a la enumeracion
     */
    private TypeOfSize convertType(String type){
        switch (type.toLowerCase()){
            case "medium" : return  TypeOfSize.medium;
            case "big" : return TypeOfSize.big;
        }

        return TypeOfSize.small;
    }

    /**
     * Getter para la propiedad vaccineNotify
     * @return boolean con el resultado de la propiedad
     */
    public boolean isVaccineNotify() {
        return vaccineNotify;
    }

    /**
     * Setter para la propiedad vaccineNotify
     * @param vaccineNotify Boleano que representa la activación / desactivación de las notificaciones
     */
    public void setVaccineNotify(boolean vaccineNotify) {
        this.vaccineNotify = vaccineNotify;
    }

    /**
     * Getter para la propiedad vaccineDay
     * @return int Valor que representa el día de vacunación
     */
    public int getVaccineDay() {
        return vaccineDay;
    }

    /**
     * Setter para la propiedad vaccineDay
     * @param vaccineDay Valor entero que representa el día de vacunación
     */
    public void setVaccineDay(int vaccineDay) {
        this.vaccineDay = vaccineDay;
    }

    /**
     * Getter para el mes de vacunación
     * @return int Valor que represnta el mes de vacunación
     */
    public int getGetVaccineMonth() {
        return vaccineMonth;
    }

    /**
     * Setter para la propiedad vaccineMonth
     * @param vaccineMonth Valor que represnta el mes de vacunación
     */
    public void setGetVaccineMonth(int vaccineMonth) {
        this.vaccineMonth = vaccineMonth;
    }

    @Override
    public String toString() {
        return "Pet{" +
                "ID=" + ID +
                ", name='" + name + '\'' +
                ", breed='" + breed + '\'' +
                ", birthday='" + birthday + '\'' +
                ", weight=" + weight +
                ", type=" + type +
                ", vaccineNotify=" + vaccineNotify +
                ", vaccineDay=" + vaccineDay +
                ", vaccineMonth=" + vaccineMonth +
                '}';
    }
}

File: clientes/android/app/src/main/java/com/lopez/guillen/dogfeeder/model/Pet.kt
package com.lopez.guillen.dogfeeder.model

import java.util.*

/**
 * Clase Pet
 * Clase encargada de modelar a la mascota asociada a DOG-FEEDER
 * @param id Identificador de la mascota
 * @param name Nombre de la mascota
 * @param breed Raza de la mascota
 * @param birthday Fecha de nacimiento de la mascota
 * @param weight Peso de la mascota
 * @param type Tipo de raza de la mascota
 * @param vaccineNotify Activación / desactivación para la notificación de vacunación anual
 * @param vaccineDay Día de vacunación
 * @param vaccineMonth Mes de vacunación
 */
class Pet(val id: Int,val name: String,val breed: String, val birthday : String, val weight : Double,
          val type : String, val vaccineNotify : Boolean, val vaccineDay : Int, val vaccineMonth : Int) {

    /**
     * Método getTypeOfSize
     * Método encargado de devolver un valor entero en función del tipo (raza) asociado a la mascota
     * @return Int Valor entero de la posicón en el listado spinner
     */
    fun getTypeOfSize() : Int{
        return when (type){
            "medium" -> 1
            "big" -> 2
            else -> 0
        }
    }


    /**
     * Método getYearOrDefault
     * Devuelve el año extraido de la fecha de nacimento y en caso de esta vacío o no cumplir con la dimensión, obtener
     * desde Calendar el año en curso.
     * @return Int Año de nacimiento
     */
    fun getYearOrDefault() : Int{
        var aDate = birthday.split("-")

        return  if(aDate.isEmpty() || aDate.size < 3){
            Calendar.getInstance().get(Calendar.YEAR)
        } else  aDate[0].toInt()
    }


    /**
     * Método getFromDate
     * Método encargado de obtener de la fecha de nacimiento en base a la cadena que llega como parametro, la parte
     * correspondiente al split de la fecha. [Si partOfYear == "YEAR" entonces return aDate[0].toInt]
     * @param partOfYear Cadena de caracteres
     * @return Int Valor correspondiente a dia, mes o año de la fecha de nacimiento
     */
    fun getFromDate(partOfYear : String) : Int {
        var aDate = birthday.split("-")

        return if(aDate.isNotEmpty() && aDate.size == 3){
            when(partOfYear){
                "YEAR" -> aDate[0].toInt()
                "MONTH" -> aDate[1].toInt() -1
                else -> aDate[2].toInt()
            }
        }else{
            val calendar = Calendar.getInstance()
            when(partOfYear){
                "YEAR" -> calendar.get(Calendar.YEAR)
                "MONTH" -> calendar.get(Calendar.MONTH)
                else -> calendar.get(Calendar.DAY_OF_MONTH)
            }
        }
    }


    override fun toString(): String {
        return "Pet(id=$id, name='$name', breed='$breed', birthday='$birthday', weight=$weight, type='$type', vaccineNotify=$vaccineNotify, vaccineDay=$vaccineDay, vaccineMonth=$vaccineMonth)"
    }
}